### DISCUSSION TITLE
Sync Multiple MultiViewer instances over network

### DISCUSSION THREAD

**Discussion by ExtremTechniker** [f1multiviewer/issue-tracker#486] [Feature requests]
Like the title says.

Option for manual master, slave sync. Slave will choose his master instance, via service discovery. Master has to enable this feature.

Option 2, choose master which will broadcast to the network with a group identifier and tick an option, on the slave instance choose which group to listen to.

Additional Feature:

Control slave nodes window placement from master node

**Comment by stephendaley185** [comment]
Really need this!

**Comment by jeppeolesen** [comment]
Hey @ExtremTechniker @stephendaley185 

MultiViewer has a Graph API that is exposed locally when the app is open - it has some controls for controlling player positions using the `playerSetBounds` mutation, as well as mutations for opening and closing players using `playerCreate` and `playerDelete`. You can access it by opening MultiViewer and going to http://localhost:10101/api/graphql in your browser of choice. So you could set up some network controls by building out an intermediary API to funnel commands from a main node to one or more child nodes, at least for the window placements and which players to open and close.

It's also possible to set an open play to sync other players to, using the `playerSync` mutation, but it only works with open players on the same device. It's not currently possible to set the target latency via the API.

But if you're up for it, it might be a fun thing to build as an integration for MultiViewer.

**Comment by LapsTimeOFF** [comment]
Also note that if the 2+ computers clocks are correctly synced (you can check that in the troubleshooting page), the live sync that we provide should already sync the devices pretty accurately.
